### ISSUE TITLE
Lines are thicker than 1px on Windows Dx12

### ISSUE THREAD

**Issue by kjarosh** [gfx-rs/wgpu#5830]
**Description**

When drawing lines using `PrimitiveTopology::LineList` or `PrimitiveTopology::LineStrip`, they are 1px thick on `Vulkan` and `Gl`, but on `Dx12` they are thicker than 1px when MSAA sample count is greater than 1.

I suspect this has something to do with their [rasterization rules](https://learn.microsoft.com/en-us/windows/win32/direct3d11/d3d10-graphics-programming-guide-rasterizer-stage-rules?redirectedfrom=MSDN#multisample-anti-aliasing-rasterization-rules):

> Rasterization rules for primitives are, in general, unchanged by multisample antialiasing, except:
> [...]
>   A line is treated as a rectangle made up of two triangles, with a line width of 1.4.

**Repro steps**

Draw a line on `Dx12` using `PrimitiveTopology::LineList` or `PrimitiveTopology::LineStrip` with `MultisampleState.count > 1`.

**Expected vs observed behavior**

Expected: lines always have 1px.

Observed: lines have 1px for `Vulkan`, `Gl`, `Dx12` without MSAA, but they are thicker than 1px for `Dx12` with MSAA.

**Extra materials**

| `Dx12` with MSAA on Windows | `Vulkan` with MSAA on Linux |
| --------------------------- | --------------------------- |
| <image src="https://github.com/gfx-rs/wgpu/assets/1507072/e592c938-e8d3-48a4-b398-7b9440253bc6" width="200"/> | <image src="https://github.com/gfx-rs/wgpu/assets/1507072/6e04bfe4-c544-49e2-ace2-b557f6a736a5" width="200"/> |

**Platform**

I actually do not use Windows at all, but unfortunately our tests at Ruffle started failing on the Windows runner when I started using `PrimitiveTopology::LineStrip` (https://github.com/ruffle-rs/ruffle/pull/16619, [failing job](https://github.com/ruffle-rs/ruffle/actions/runs/9464982211/job/26073492650)). I have confirmed that it in fact does look like that on a Windows machine.

**Comment by cwfitzgerald** [comment]
This probably needs to be bumped up to webgpu as it's unclear what we're supposed to do here - I suspect there's not a ton we can do about this, without extremely expensive polyfills.

**Comment by Diggsey** [comment]
Couldn't you just set `MultisampleEnable` to `FALSE` in the rasterizer state?

https://learn.microsoft.com/en-us/windows/win32/api/d3d11/ns-d3d11-d3d11_rasterizer_desc

> MultisampleEnable
> 
> Type: [BOOL](https://learn.microsoft.com/en-us/windows/desktop/WinProg/windows-data-types)
> 
> Specifies whether to use the quadrilateral or alpha line anti-aliasing algorithm on multisample antialiasing (MSAA) render targets. Set to TRUE to use the quadrilateral line anti-aliasing algorithm and to FALSE to use the alpha line anti-aliasing algorithm. For more info about this member, see Remarks.

**Comment by cwfitzgerald** [comment]
Yeah let's do that and it would be great to have a tea) test for this
